tests/components/minio/test_minio.py

```diff
@@ -0,0 +1,202 @@
+"""Tests for Minio Hass related code."""
+import asyncio
+import json
+from unittest.mock import MagicMock
+
+import pytest
+from asynctest import patch, call
+
+from homeassistant.components.minio import QueueListener, DOMAIN, CONF_HOST, \
+    CONF_PORT, CONF_ACCESS_KEY, CONF_SECRET_KEY, CONF_SECURE, CONF_LISTEN, \
+    CONF_LISTEN_BUCKET
+from homeassistant.core import callback
+from homeassistant.setup import async_setup_component
+
+from tests.components.minio.common import TEST_EVENT
+
+
+@pytest.fixture(name='minio_client')
+def minio_client_fixture():
+    """Patch Minio client."""
+    with patch(
+            'homeassistant.components.minio.minio_helper.Minio'
+    ) as minio_mock:
+        minio_client_mock = minio_mock.return_value
+
+        yield minio_client_mock
+
+
+@pytest.fixture(name="minio_event_response")
+def minio_event_response_fixture():
+    """Patch helper function for minio notification stream."""
+    with patch(
+        'homeassistant.components.minio'
+        '.minio_helper.get_minio_notification_response'
```

Comment:
We shouldn't patch our function, but we should patch the minio client function that `get_minio_notification_response` uses.